modules/shared_state.py

```diff
@@ -76,8 +77,12 @@ def skip(self):
         log.info("Received skip request")
 
     def interrupt(self):
-        self.interrupted = True
-        log.info("Received interrupt request")
+        if shared.opts.interrupt_after_current and self.job_count > 1:

```

Comment:
About the `self.job_count > 1` condition, I think that's the intuitive behavior that:

- Interrupt right away when generating single image
- Interrupt on next when doing batch
- To interrupt right away in batch, the user could do "Interrupt" and then "Skip".